common/src/main/java/com/mapbox/vision/common/view/BaseTeaserActivity.kt

```diff
@@ -0,0 +1,490 @@
+                                    setImageResource(imageResource)
+                                    translationY = 0f
+                                    alpha = 1f
+                                    scaleX = 1f
+                                    scaleY = 1f
+                                    speed_limit_next.hide()
+                                }
+
+                                override fun onAnimationCancel(animation: Animator?) {}
+
+                                override fun onAnimationStart(animation: Animator?) {}
+                            }
+                        )
+                        .setInterpolator(AccelerateDecelerateInterpolator())
+                        .start()
+                }
+
+                if (roadRestrictions.speedLimits.car.max != 0f) {
+                    speed_limit_next.apply {
+                        translationY = -speedLimitTranslation
+                        setImageResource(imageResource)
+                        show()
+                        animate().translationY(0f)
+                            .setDuration(500L)
+                            .setInterpolator(AccelerateDecelerateInterpolator())
+                            .start()
+                    }
+                } else {
+                    speed_limit_next.hide()
+                }
+            }
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        soundsPlayer = SoundsPlayer(this)
+    }
+
+    override fun initViews() {
+        setContentView(R.layout.activity_main)
+    }
+
+    override fun onPermissionsGranted() {
+        isPermissionsGranted = true
+
+        signSize = resources.getDimension(R.dimen.dp64).toInt()
+        lineHeight = resources.getDimension(R.dimen.dp40).toInt()
+        margin = resources.getDimension(R.dimen.dp8).toInt()
+
+        back.setOnClickListener { onBackClick() }
+        segm_container.setOnClickListener { setAppMode(AppMode.Segmentation) }
+        sign_detection_container.setOnClickListener { setAppMode(AppMode.Classification) }
+        det_container.setOnClickListener { setAppMode(AppMode.Detection) }
+        distance_container.setOnClickListener { setAppMode(AppMode.Safety) }
+        safety_mode_container.hide()
+        line_detection_container.setOnClickListener { setAppMode(AppMode.Lanes) }
+
+        root.setOnLongClickListener {
+            if (fps_performance_view.visibility == View.GONE) {
+                fps_performance_view.show()
+            } else {
+                fps_performance_view.hide()
+            }
+            return@setOnLongClickListener true
+        }
+        fps_performance_view.hide()
+
+        tryToInitVisionManager()
+
+        object_mapping_button_container.setOnClickListener { view1Click.invoke() }
+        ar_navigation_button_container.setOnClickListener { view2Click.invoke() }
+
+        when (getAppType()) {
+            AppType.Teaser -> {
```

Comment:
What is the reason to handle this logic here, instead of just adding some logic in specific activities?

Reply:
I use the same layout for 2 apps, just change the last 2 items (icon, text, click listeners).